### ISSUE TITLE
Check if the cluster mode is consistent between master and slaves when replication

### ISSUE THREAD

**Issue by git-hulk** [apache/kvrocks#1411]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Motivation

Currently, we didn't check the cluster status between the master and slaves, and the slave may crash if the master enabled the cluster node but not enabled in slaves, and vice versa.

### Solution

Check the cluster status when replication.

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by zevin02** [comment]
please assign to me

**Comment by PragmaTwice** [comment]
> please assign to me

@git-hulk is self-assigned now, so it may not available for you.

**Comment by git-hulk** [comment]
Yes, it needs to add the cluster status into the replication process without breaking changes, so I assigned it to myself. Thanks for your attention. @zevin02

**Comment by uds5501** [comment]
What are the deliverables / check points for this task? I can think of 2 broad ones :
- When trying to toggling cluster mode on or off in master / slave, verify if both are on same mode or not, if not then maybe master can propogate the cluster mode `on` to slave as well (additional thoughts: this may break the cluster configs in other nodes of the cluster as they will be unaware of this new slave)

- When a master node is part of the cluster and a new instance is going to replicate this master, then it needs to be configured in such a way that it's part of the cluster as well.

LMK if I misunderstood these deliverables.

**Comment by git-hulk** [comment]
Hi @uds5501 Thanks for your attention.

Like Redis, Kvrocks can run in the master-replica or cluster mode(by enabling the cluster-enabled in kvrocks.conf), we would like to check if it configured the cluster mode consistently when setup the replication.
